### PR TITLE
formula_dir update

### PR DESCRIPTION
**Pull request by AnastasiaSulyagina**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
Required for `brew update` to work properly with deleted and moved casks (for context see [#15603](https://github.com/caskroom/homebrew-cask/issues/15603))

cc @MikeMcQuaid

### CODE REVIEW

File: Library/Homebrew/tap.rb [hunk 1]
```diff
@@ -299,7 +299,7 @@ def custom_remote?
 
   # path to the directory of all {Formula} files for this {Tap}.
   def formula_dir
-    @formula_dir ||= [path/"Formula", path/"HomebrewFormula", path].detect(&:directory?)
+    @formula_dir ||= [path/"Formula", path/"HomebrewFormula", path/"Casks", path].detect(&:directory?)
```
Comment (MikeMcQuaid) [inline Library/Homebrew/tap.rb]: Probably worth adding a new function `cask_dir` and use that instead?

File: Library/Homebrew/tap.rb [hunk 2]
```diff
@@ -302,6 +303,11 @@ def formula_dir
     @formula_dir ||= [path/"Formula", path/"HomebrewFormula", path].detect(&:directory?)
   end
 
+  # path to the directory of all casks for caskroom/cask {Tap}.
+  def cask_dir
+    @cask_dir ||= [path/"Casks", path].detect(&:directory?)
```
Comment (xu-cheng) [inline Library/Homebrew/tap.rb]: This should just be `path/Casks` as I don't remember cask supporting put cask file in the top directory, right?

File: Library/Homebrew/tap.rb [hunk 3]
```diff
@@ -317,7 +323,7 @@ def formula_files
   def formula_file?(file)
     file = Pathname.new(file) unless file.is_a? Pathname
     file = file.expand_path(path)
-    file.extname == ".rb" && file.parent == formula_dir
+    file.extname == ".rb" && (file.parent == formula_dir || file.parent == cask_dir)
```
Comment (xu-cheng) [inline Library/Homebrew/tap.rb]: 👎 I suggest you to create a new method `cask_file?`

File: Library/Homebrew/tap.rb [hunk 4]
```diff
@@ -561,6 +567,14 @@ def formula_dir
   end
 
   # @private
+  def cask_dir
+    @cask_dir ||= begin
+      self.class.ensure_installed!
+      super
+    end
+  end
```
Comment (xu-cheng) [inline Library/Homebrew/tap.rb]: Core tap, i.e.`homebrew/core` doesn't have casks. So you don't need this method.

**Comment by MikeMcQuaid** [comment]
Thanks again @AnastasiaSulyagina! 🎉

**Pull request closed without merging**